internal/jobs/launch_instance_aws.go

```diff
@@ -70,6 +73,10 @@ func handleLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArg
 	if err != nil {
 		return fmt.Errorf("cannot get pubkey by id: %w", err)
 	}
+	pkr, err := pkD.UnscopedGetResourceBySourceAndRegion(ctx, pk.ID, args.SourceID, args.Region)
+	if err != nil {
+		return fmt.Errorf("unable to fetch pubkey resource: %w", err)
+	}
```

Comment:
Isn't more straightforward to simply fetch this before the job and pass both pubkey and its resources? I mean, DB lookups are fine to be done outside of background job, they are fast. It's AWS API calls what should be performed in a job.

Reply:
I guess, but what do you think by passed to the job? the whole model in args? thinking that feels unsafe to me to marshal and unmarshal it for no reason - it feels better to just fetch it from the DB in the job shrug

Reply:
Right.